pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -89,6 +90,35 @@ func (pl *VolumeBinding) Name() string {
 	return Name
 }
 
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
+func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEvent {
+	events := []framework.ClusterEvent{
+		// Pods may fail because of missing or mis-configured storage class
+		// (e.g., allowedTopologies, volumeBindingMode), and hence may become
+		// schedulable upon StorageClass Add or Update events.
+		{Resource: framework.StorageClass, ActionType: framework.Add | framework.Update},
+		// We bind PVCs with PVs, so any changes may make the pods schedulable.
+		{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update},
+		{Resource: framework.PersistentVolume, ActionType: framework.Add | framework.Update},
+		// Pods may fail to find available PVs because the node labels do not
+		// match the storage class's allowed topologies or PV's node affinity.
+		// A new or updated node may make pods schedulable.
+		{Resource: framework.Node, ActionType: framework.Add | framework.Update},
+		// We rely on CSI node to translate in-tree PV to CSI.
+		{Resource: framework.CSINode, ActionType: framework.Add | framework.Update},
+	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.CSIStorageCapacity) {
+		// When CSIStorageCapacity is enabled, pods may become schedulable
+		// on CSI driver & storage capacity changes.
+		events = append(events, []framework.ClusterEvent{
+			{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update},
+			{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update},
```

Comment:
We don't discover core API res events automatically on the event handler side (we do for CRD), so need to add cases for CSIDriver and CSIStorageCapactiy in eventhandlers.go#addAllEventHandlers().

(don't need to check feature gate in eventhandlers.go#addAllEventHandlers() as if it's disabled, it won't show up in the eventsMap)

BTW: please take a second look to ensure all other "new" core events are all covered in addAllEventHandlers().

Reply:
done in latest commit